src/mmap.rs

```diff
@@ -74,6 +74,7 @@ pub unsafe fn mremap_helper(old_address: *mut c_void,
     syscall!(MREMAP, old_address, old_len, new_len, flags, new_address) as *mut c_void
 }
 
+#[cfg_attr(all(feature = "weak", not(windows), not(target_os = "macos")), linkage = "weak")]
```

Comment:
Is there a reason we can't just specify `#cfg[(linkage = "weak")]`? I don't see a reason why we'd compile for Windows or Mac, and any Unix that hypothetically gets supported would have to support weak linkage as well.

Might be missing something though.